File: server/dotnet/sdk-dotnet-example/Business/Services/HostedCheckoutService.cs
using Business.DTOs.CreateHostedCheckouts;
using Business.DTOs.GetPaymentByHostedCheckoutId;
using Business.Interfaces.SDKClients;
using Business.Interfaces.Services;

namespace Business.Services;

public class HostedCheckoutService(IHostedCheckoutClient hostedCheckoutClient)
    : IHostedCheckoutService
{
    public async Task<CreateHostedCheckoutResponseDto> CreateHostedCheckoutSessionsAsync(
        CreateHostedCheckoutRequestDto request)
    {
        CreateHostedCheckoutResponseDto responseDto =
            await hostedCheckoutClient.CreateHostedCheckoutSessionAsync(request);

        responseDto.Amount = request.Amount;
        responseDto.Currency = request.Currency;

        return responseDto;
    }

    public async Task<GetPaymentByHostedCheckoutIdResponseDto?> GetPaymentByHostedCheckoutIdAsync(
        string id)
    {
        return await hostedCheckoutClient.GetPaymentByHostedCheckoutIdAsync(id);
    }
}

File: server/ruby/app/business/services/create_hosted_checkout/hosted_checkout_service.rb
module Business
  module Services
    module HostedCheckout
      class HostedCheckoutService
        def initialize(hosted_checkout_client:)
          @hosted_checkout_client = hosted_checkout_client
        end

        def create_hosted_checkout(request_dto)
          response_dto = @hosted_checkout_client.create_hosted_checkout(request_dto)

          response_dto.amount = request_dto.amount
          response_dto.currency = request_dto.currency

          response_dto
        end

        def get_payment_by_hosted_checkout_id(id)
          @hosted_checkout_client.get_payment_by_hosted_checkout_id(id)
        end
      end
    end
  end
end